manifests/env.pp

```diff
@@ -6,21 +6,29 @@
 # parameter
 
 define poudriere::env (
-  $ensure           = 'present',
-  $makeopts         = [],
-  $makefile         = nil,
-  $version          = '10.0-RELEASE',
-  $arch             = 'amd64',
-  $jail             = $name,
-  $paralleljobs     = $facts['processors']['count'],
-  $pkgs             = [],
-  $pkg_file         = nil,
-  $pkg_makeopts     = {},
-  $pkg_optsdir      = nil,
-  $portstree        = 'default',
-  $cron_enable      = false,
-  $cron_always_mail = false,
-  $cron_interval    = { minute => 0, hour => 0, monthday => '*', month => '*', weekday => '*' },
+  Enum['present', 'absent'] $ensure           = 'present',
+  Array[String]             $makeopts         = [],
+  Optional[String::Abspath] $makefile         = undef,
+  String                    $version          = '10.0-RELEASE',
+  String                    $arch             = 'amd64',
+  String                    $jail             = $name,
+  Integer                   $paralleljobs     = $facts['processors']['count'],
+  Array[String[1]]          $pkgs             = [],
+  Optional[Stdlib::Abspath] $pkg_file         = undef,
+  Hash                      $pkg_makeopts     = {},
+  Optional[Stdlib::Abspath] $pkg_optsdir      = undef,
+  String[1]                 $portstree        = 'default',
+  Boolean                   $cron_enable      = false,
+  Boolean                   $cron_always_mail = false,
+  Struct[
+    {
+      minute   => Optional[Variant[Integer, String]],
+      hour     => Optional[Variant[Integer, String]],
+      monthday => Optional[String],
+      month    => Optional[String],
+      week     => Optional[String],
+    }
+  ]                        $cron_interval    = { minute => 0, hour => 0, monthday => '*', month => '*', weekday => '*' },
```

Comment:
Perhaps this should be extracted to a separate type?

Reply:
Indeed!  I wanted to do it at some point and forgot about it.